### PR TITLE
Add log enrichment use case to Data Prepper

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Adds log enrichment with Data Prepper to open source documentation

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
Hi @lizsnyder: Please review this use case carryover to Data Prepper at your availability. Thank you :)

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (lizsnyder) [inline _data-prepper/common-use-cases/log-enrichment.md:44]: Probably "URL that can be paginated" to avoid the build error.

**Review by lizsnyder — APPROVED** [review]
LGTM, approved.

**Review by Naarcha-AWS — APPROVED** [review]

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:12]:
```suggestion
- Filtering.
```

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:44]:
```suggestion
Log data often includes strings of key-value pairs. For example, if a user queries a URL that can be paginated, the HTTP logs might contain the following HTTP query string:
```

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:98]: "existing `date` and `time` entries"?

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:111]:
```suggestion
Consider the following incoming event:
```

Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:111]: Removed "For example" here to avoid repetition from the preceding sentence.

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:121]: "a new key named `key_three`"?

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:134]:
```suggestion
The various [Mutate string]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/mutate-string/) processors offer tools that you can use to manipulate strings in incoming data. For example, if you need to split a string into an array, you can use the `split_string` processor:
```

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:151]:
```suggestion
The [List-to-map]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/list-to-map/) processor, which is one of the Mutate event processors, converts a list of objects in an event to a map.
```

Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:151]: Can "Mutate event" be formatted so as to avoid any mid-sentence capitalization confusion? Last phrase: "converts a list of event objects to a map"?

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:190]:
```suggestion
to a map:
```

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:203]:
```suggestion
As another example, consider an incoming event with the following structure:
```

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:190]: I'm not a huge fan of splitting sentences over multiple lines, but I'll defer to you if this is best practice.

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:283]:
```suggestion
In many cases, you may want to flatten the array for each key. In these situations, you must choose only one object to retain. The processor offers a choice of either first or last.
```

Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:283]: Should this sentence end with a colon?

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:324]: Should we define International Organization for Standardization (ISO)?

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:348]:
```suggestion
It converts the event to the following format:
```

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:360]: Can "Date processor" be formatted so as to avoid any mid-sentence capitalization confusion?

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:374]: Can "Mutate string" be formatted so as to avoid any mid-sentence capitalization confusion?

Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:374]:
```suggestion
The [Substitute string]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/substitute-string/) processor (which is one of the Mutate string processors) lets you derive a punctuation pattern from incoming events. In the following example pipeline, the processor will scan incoming Apache log events and derive punctuation patterns from them:
```

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:360]:
```suggestion
The Date processor can generate timestamps for incoming events if you specify `@timestamp` for the `destination` option:
```

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (natebower) [inline _data-prepper/common-use-cases/log-enrichment.md:324]:
```suggestion
The [Date]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/date/) processor parses the timestamp key from incoming events by converting it to ISO 8601 format:
```

**Review by natebower — COMMENTED** [review]
@vagimeli Please see my comments and changes and let me know if you have any questions. Thanks!

File: _data-prepper/common-use-cases/log-enrichment.md
Comment (vagimeli) [reply _data-prepper/common-use-cases/log-enrichment.md]: In researching the naming convention "mutate event processor," I didn't come across it as a proper name. Instead, it appears as a common name for processors that convert an event.

**Review by natebower — APPROVED** [review]
LGTM if all changes/comments were addressed.

**Comment by vagimeli** [comment]
@lizsnyder This PR has been merged. Tagging you for visibility.

**Pull request merged**
